edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -72,19 +86,87 @@ def _add_missing_jwt_permission_classes(self, view_class):
             view_class.permission_classes += tuple(classes_to_add)
 
     def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
-        # Views as functions store the view's class in the 'view_class' attribute.
-        # Viewsets store the view's class in the 'cls' attribute.
-        view_class = getattr(
-            view_func,
-            'view_class',
-            getattr(view_func, 'cls', view_func),
-        )
+        view_class = _get_view_class(view_func)
 
         view_authentication_classes = getattr(view_class, 'authentication_classes', tuple())
-        if self._includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
+        if _includes_base_class(view_authentication_classes, BaseJSONWebTokenAuthentication):
             self._add_missing_jwt_permission_classes(view_class)
 
 
+class JwtAuthWithLoginRequiredMiddleware(MiddlewareMixin):
```

Comment:
Consider renaming this to: `JwtRedirectToLoginIfUnauthenticatedMiddleware`.  Such a name may make it explicit on what it enables.

Reply:
Done.